src/dataflow/source/kafka.rs

```diff
@@ -196,6 +198,9 @@ where
 
                 while let Some(message) = next_message {
                     let partition = message.partition;
+                    if message.offset < start_offset {
```

Comment:
This not necessarily an error. Kafka brokers disconnect and reconnect. When they do, clients restart processing the stream from scratch. This is the reason for the call` if offset <= last_processed_offset` below. It might be better to modify the error message there to something other than "Duplicate" that captures this?